interview-questions.md

```diff
@@ -6,35 +6,63 @@ First, without external resources. Challenge yourself to answer from memory.
+
+celebrity_crush = Hash.new 
+
+, or implicitly by just creating the hash with a name and a list of key-value pairs separated by commas,
+
+celebrity_crush = {name: "Nicole Behaire", occupation: "actress", favorite_attribute: "smile"}
+
 
-Your answer:
 
-Researched answer:
 
 2. What is a gem?
 
-Your answer:
+Your answer: A gem is a package manager the works for Ruby much like how yarn works for JavaScript in that is contains code that can be shared amongst developers.  
 
-Researched answer:
+Researched answer: A gem in ruby is a an open source library of code that is meant to be shared and used by developers within their programs. Their are a great number of gems, Rspec and Rails are examples of 2 gems. These libraries canbe found public repositories like RubyGems.org. and GitHub.
 
 3. What is Ruby on Rails?
 
-Your answer:
+Your answer: Ruby on Rails is a open-source framework that works with the ruby language. The way rails is structured is based off of MVC structure short for model, veiw, controller. The model being the database, the veiw being all on the content being veiwed by users (HTML, CSS, JavaScript), and the controller the director tell all the data where to go. 
 
-Researched answer:
+Researched answer: Ruby on Rails is a MVC framework used for web applications. MVC stands for Model Veiw Controller and is a way to organize code. The Model part for MVC is the database layer and holds data like lists and corresponding attributes. The Veiw of MVC is the code for user interface so this is all of the HTML, JavaScript, and CSS. The controller is the brains of the framework. It takes the inputs from the user interaction and dictates what should happen with the view and model portions.
 
-4. What is a relational database? Are there other kinds of databases?
+4. What is a relational database? Are there other kinds of databases? 
 
-Your answer:
+Your answer: relational datatbases are similar to excel spread sheets. They hold information in different rows and columns. 
 
-Researched answer:
+Researched answer: Relational databases are a type of database that can store data in a table much like an excel spreadsheet. Information is stored in each row of the table and gieven a unique identifier also know as a primary key. With this given key relational databases can access data that is related to eachother. 
 
 5. What are primary keys? Why are they important?
 
-Your answer:
+Your answer: Primary keys are unique identifiers for every instances within a model. These are important when being able to call on specific instances
 
-Researched answer:
+Researched answer: Primary Keys occupy the first colum of the model. They are an identifier for every row or instance within the model table. They are important because they can be used to access related information within relational databases. 
```

Comment:
Awesome job on these answers.  I love your examples and your answers are becoming more conversational.  Really good work!